android/sdk/src/main/java/app/organicmaps/sdk/sound/AudioFocusManagerImpl.java

```diff
@@ -0,0 +1,40 @@
+package app.organicmaps.sdk.sound;
+
+import static android.media.AudioManager.AUDIOFOCUS_GAIN_TRANSIENT_MAY_DUCK;
+
+import android.content.Context;
+import android.media.AudioFocusRequest;
+import android.media.AudioManager;
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+@RequiresApi(26)
+final class AudioFocusManagerImpl extends AudioFocusManager
+{
+  @NonNull
+  private final AudioFocusRequest mAudioFocusRequest;
+
+  public AudioFocusManagerImpl(@NonNull Context context, @NonNull OnAudioFocusLost onAudioFocusLost)
+  {
+    super(context, onAudioFocusLost);
+    mAudioFocusRequest = new AudioFocusRequest.Builder(AUDIOFOCUS_GAIN_TRANSIENT_MAY_DUCK)
+                             .setAudioAttributes(AUDIO_ATTRIBUTES)
+                             .setOnAudioFocusChangeListener(this::onAudioFocusChange)
+                             .build();
+  }
+
+  @Override
+  public boolean requestAudioFocus()
+  {
+    final int requestResult = mAudioManager.requestAudioFocus(mAudioFocusRequest);
+    mPlaybackAllowed = requestResult == AudioManager.AUDIOFOCUS_REQUEST_GRANTED;
```

Comment:
Print a warning or error when it failed, with a reason of the failure?

Reply:
There is no reason, unfortunately, only `AUDIOFOCUS_REQUEST_FAILED` or `AUDIOFOCUS_REQUEST_GRANTED`